torch/csrc/jit/backends/backend_debug_handler.h

```diff
@@ -125,31 +126,17 @@ using BackendDebugInfoMapType =
 class TORCH_API BackendDebugInfoRecorder {
  public:
   BackendDebugInfoRecorder() = default;
-
   int64_t getNextDebugHandle(const Node* node);
   // Reason this is not done as RAII is that work done in stopRecording
   // can throw, and throwing with dtor will call terminate and thus voids any
   // exception catching at a higher level.
   BackendDebugInfoMapType stopRecording();
+  NodeToDebugHandle generate_debug_handles(const std::shared_ptr<Graph>& graph);
```

Comment:
See my comment below (in the preprocess function) about having a separate object to handle this which the recorder can produce and pass to preprocess functions.